compiler/test/dotty/tools/dotc/profile/ChromeTraceTest.scala

```diff
@@ -89,5 +89,18 @@ class ChromeTraceTest:
           assertTrue(ts4 >= ts3 + 8.millis.toMicros)
         case _ => fail("unreachable")
       }
+    }
+  }
+
+  private def sleep(duration: FiniteDuration): Unit = {
+    // A bit of additional precautions to ensure we don't continue execution to early
+    // Both LockSuppport and Thread.sleep can return earlier then expected (depending on OS)
+    var remainingNanos = duration.toNanos
```

Comment:
Not sure if this is just a matter of personal taste or if it might make some difference in the actual behaviour:
Why not to compute the expected end time of sleep before the lop and then compare it with `System.nanoTime` in the loop?

Reply:
It might behave differently. If we just sleep again using `duration` then in the worst case, we can sleep for 1.99x of expected time. To prevent that, we'd need to calculate the remaining time after pre-mature wake up anyway. 

1 possible improvement could be potentially usage of `TimeUnit.sleep(ofUnit: Long)` but that would probably make it more complicate. Previously I though it's available only since JDK 9

Reply:
Of course I didn't mean sleeping twice with the same duration. I was thinking of something like
```scala
  private def sleep(duration: FiniteDuration): Unit = {
    // A bit of additional precautions to ensure we don't continue execution to early
    // Both LockSuppport and Thread.sleep can return earlier then expected (depending on OS)
    // var remainingNanos = duration.toNanos
    var currentNanos = System.nanoTime()
    val deadline = currentNanos + duration.toNanos
    while
      currentNanos = System.nanoTime()
      currentNanos < deadline
    do
      val remainingNanos = deadline - currentNanos
      val millis = NANOSECONDS.toMillis(remainingNanos)
      Thread.sleep(millis, (remainingNanos % 1.millis.toNanos).toInt)
  }
```
But maybe this doesn't make that much of a difference